back/engines/commercial/analysis/app/lib/analysis/summarization_plan.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module Analysis
+  # The plan is the output of the #plan method on the summarization method. It
+  # describes how the summarization should be done, mainly what method and which
+  # LLM to use
+  class SummarizationPlan
+    include ActiveModel::API
+    IMPOSSIBLE_REASONS = [:too_many_inputs]
+
+    attr_accessor :summarization_method_class, :llm, :truncate_values, :include_id, :shorten_labels, :accuracy, :impossible_reason
+
+    with_options unless: -> { impossible_reason } do
+      validates :summarization_method_class, inclusion: { in: SummarizationMethod::Base::SUMMARIZATION_METHOD_CLASSES }
+      validates :llm, inclusion: { in: SummarizationMethod::Base::LLMS }
+      validates :truncate_values, numericality: { only_integer: true }, allow_blank: true
+      validates :include_id, inclusion: { in: [true, false] }, allow_blank: true
+      validates :shorten_labels, inclusion: { in: [true, false] }, allow_blank: true
+      validates :accuracy, numericality: { in: 0..1 }, allow_blank: true
+    end
+
+    validates :impossible_reason, inclusion: { in: IMPOSSIBLE_REASONS }, allow_blank: true
```

Comment:
It's never called indeed, these are not used.

I was thinking about persisting this first, then decided against it, and figured I might as well keep them as a form of documentation. But I do see it's confusing. I'll add a comment to explain.

Reply:
Maybe something like this could be used https://dry-rb.org/gems/dry-types/1.7/
We could also use it later for command objects (we discussed how difficult it is to support these big hashes).

Reply:
Would be cool for us to look into types more, would definitely look at all the options before we head there. They're slowly maturing in the ecosystem.